runtime/ops/worker_host.rs

```diff
@@ -582,19 +590,49 @@ fn op_host_terminate_worker(
   Ok(())
 }
 
-/// Try to remove worker from workers table - NOTE: `Worker.terminate()`
-/// might have been called already meaning that we won't find worker in
-/// table - in that case ignore.
-fn try_remove_and_close(state: Rc<RefCell<OpState>>, id: WorkerId) {
+enum WorkerChannel {
+  Ctrl,
+  Messages,
+}
+
+/// Close a worker's channel. If this results in both of a worker's channels
+/// being closed, the worker will be removed from the workers table.
+fn close_channel(
```

Comment:
The name of this function (and the two added fields to `WorkerThread`) needs bikeshedding, since it doesn't really close the underlying channel – it just marks it as closed, and cleans up the worker if both of its channels are closed.